### PR TITLE
Fix #271 - Support multiple jwt issuers

### LINKED ISSUE #271: Support Multiple Trusted JWT Issuers for microgateway
**Description:**
In the current design of the micro-gateway, we can use the micro-gateway.conf file to configure information about one trusted issuer as below.

```
[jwtTokenConfig]
issuer="https://sts.wso2.com"
audience="http://org.wso2.apimgt/gateway"
certificateAlias="wso2apim"
trustStore.path="${ballerina.home}/bre/security/ballerinaTruststore.p12"
trustStore.password="ballerina"
```

This only allows us to validate only JWTs issued by the configured trusted JWT issuer at the micro gateway.

As an improvement it would better if we could support multiple trusted JWT issuers.

**Suggested Labels:**
Improvement

**Affected Product Version:**
2.6.0

### PR DESCRIPTION
**Pull request by Rajith90**
### Purpose
In mgw 3.0.2 version, JWT can be validated from a single trusted issuer.
With this feature we should be able to add multiple trusted jwt issuers, and micro gw will accept tokens issued by any of those trusted issuers

### Issues

Fixes #271 

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Comment by Rajith90** [comment]
Depends on : https://github.com/ballerina-platform/ballerina-lang/issues/20390

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal [hunk 1]
```diff
@@ -89,30 +89,8 @@ public function initiateGatewayConfigurations(http:ListenerConfiguration config)
...
-    } else {
-        jwtAuthHandler = new JWTAuthHandler(jwtAuthProvider);
-    }
-
-
+    //Read multiple jwt issuers
+    http:InboundAuthHandler[] multipleJwtIssuers = readMultipleJWTIssuers();
     // Initializes the key validation handler
     KeyValidationServerConfig keyValidationServerConfig = {
         url: getConfigValue(KM_CONF_INSTANCE_ID, KM_SERVER_URL, "https://localhost:9443"),
```
Comment (AmaliMatharaarachchi) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]: Please change default config values as in #948

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal [hunk 2]
```diff
@@ -158,8 +136,59 @@ public function getAuthHandlers() returns http:InboundAuthHandler[] {
...
+    http:InboundAuthHandler[] issuerArray = [];
+    string issuerId = JWT_INSTANCE_ID;
+    while (true) {
+        if(issuerCount > 0 ) {
+            issuerId = JWT_INSTANCE_ID + issuerCount.toString();
+        }
+        map<any> issuerConfig = getConfigMapValue(issuerId);
+        if (issuerConfig.length() > 0 ) {
+            jwt:JwtValidatorConfig jwtValidatorConfig = {
+                issuer: getConfigValue(issuerId, ISSUER, "https://localhost:9443/oauth2/token"),
```
Comment (AmaliMatharaarachchi) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]: Change to the default config value.

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal [hunk 3]
```diff
@@ -158,8 +136,59 @@ public function getAuthHandlers() returns http:InboundAuthHandler[] {
...
+                audience: getConfigValue(issuerId, AUDIENCE, "http://org.wso2.apimgt/gateway"),
```
Comment (AmaliMatharaarachchi) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]: Change to the default config value.

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal [hunk 4]
```diff
@@ -158,8 +136,59 @@ public function getAuthHandlers() returns http:InboundAuthHandler[] {
...
+                clockSkewInSeconds: 60,
+                trustStoreConfig: {
+                    trustStore: {
+                        path: getConfigValue(LISTENER_CONF_INSTANCE_ID, TRUST_STORE_PATH,
```
Comment (AmaliMatharaarachchi) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]: Change to the default config value.

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal [hunk 5]
```diff
@@ -158,8 +136,59 @@ public function getAuthHandlers() returns http:InboundAuthHandler[] {
...
+                        "${ballerina.home}/bre/security/ballerinaTruststore.p12"),
+                        password: getConfigValue(LISTENER_CONF_INSTANCE_ID, TRUST_STORE_PASSWORD, "ballerina")
```
Comment (AmaliMatharaarachchi) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]: Change to the default config value.

File: components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal [hunk 6]
```diff
@@ -158,8 +136,59 @@ public function getAuthHandlers() returns http:InboundAuthHandler[] {
...
+                    },
+                    certificateAlias: getConfigValue(issuerId, CERTIFICATE_ALIAS, "ballerina")
```
Comment (AmaliMatharaarachchi) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal]: Change to the default config value.

**Comment by praminda** [comment]
`jballerina` branch is no longer valid for accepting PR. I changed the base to `master`. Please resolve the conflicts

**Comment by Rajith90** [comment]
Fixed with https://github.com/wso2/product-microgateway/pull/1023

**Pull request closed without merging**